vulnerabilities/unchecked-return-values.md

```diff
@@ -52,7 +53,36 @@ A more serious version of this bug occurred in [King of the Ether](https://www.k
 
 To mitigate this vulnerability, developers should always check the return value of any call to an external contract. The `require()` function can be used to check if the call was successful and handle any errors that may occur.
 
+A caveat developers should be wary of when using the `require()` function is unexpected reverts that can cause DoS. If the developer naively decides to check for for the success or failure of the external `.send()` call like so:
+
+```solidity
+/// INSECURE
+contract Lotto {
+
+    bool public paidOut = false;
+    address public winner;
+    uint256 public winAmount;
+
+    /// extra functionality here
+
+  function sendToWinner() public {
+        require(!paidOut);
+        require(winner.send(winAmount));    // naively check success of the external call
+        paidOut = true;
+    }
+
+  function withdrawLeftOver() public {
+        require(paidOut);  // requires `paidOut` to be true
+        msg.sender.send(this.balance);
+    }
+```
+
+An attacker interacting with the `Lotto` contract from their own malicious contract and calling the `sendToWinner` function, can just implement a fallback function that reverts all payments making `paidOut` not set to true!
+
+A detailed explanation of this caveat can be found [here](https://github.com/kadenzipfel/smart-contract-vulnerabilities/blob/master/vulnerabilities/dos-revert.md)
```

Comment:
We should do a relative link to the file here, kinda like how we do links in the readme, just in case the url changes at some point